gapir/cc/interpreter.h

```diff
@@ -69,7 +70,8 @@ class Interpreter {
   // Creates a new interpreter with the specified memory manager (for resolving
   // memory addresses) and with the specified maximum stack size
   Interpreter(core::CrashHandler& crash_handler,
-              const MemoryManager* memory_manager, uint32_t stack_depth);
+              const MemoryManager* memory_manager, uint32_t stack_depth,
+              ReplayService* srv);
```

Comment:
We should probably just have an std::function or similar here, instead of passing the entire ReplayService object down, just so we can get at one method. The rest of our methods are handled through what amounts to callbacks, so we should do something similar here, instead of coupling further.

Reply:
Thanks! This is a much better code pattern! Have made another commit to address this problem.